### PR TITLE
Fully Accessible and Bootstrap-Free navbar

### PR DESCRIPTION
**Pull request by mansona**
Howdy folks 👋 

So I had a conversation with @MelSumner about the overall layout of pages using styleguide (specifically the website) and one of the blocking points for moving forward was that we are currently including bootstrap so that we can make use of the bootstrap navbar which is already quite accessible.

I have been tasked with implementing http://w3c.github.io/aria-practices/examples/menubar/menubar-1/menubar-1.html as an alternative and wow has this been a challenge 😳 

You can see quite a lot of dom manipulation going on here but I have (mostly) captured all of the functionality of the example menu bar. Here is a video demoing the functionality: https://cl.ly/2V0E0t0P0O3V

There is probably quite a lot of cleanup that will need to be done here before we actually merge this (not least the actual CSS) but I think it's a pretty good start in the implementation 😍 

Let me know if you have any thoughts or questions 👍

### CODE REVIEW

**Comment by sivakumar-kailasam** [comment]
Awesome stuff! Any chance this could be its own addon? 😄 

Curious, have you considered using https://emberobserver.com/addons/ember-keyboard for handling keyboard events?

**Comment by MelSumner** [comment]
@mansona sorry to pull the rug out- updated to ember-cli-addon-docs. Would you update when you get the chance?

**Comment by MelSumner** [comment]
> Curious, have you considered using https://emberobserver.com/addons/ember-keyboard for handling keyboard events? 
@sivakumar-kailasam we should evaluate for a11y- I remember looking at it before and noting that there were some a11y concerns but I don't remember what they were at this moment.

**Comment by MelSumner** [comment]
@mansona also we'd want to remove the letter keyboard shortcuts- screen readers already have their own uses for the letters, and we'd most likely be conflicting with that. Have you checked out the screen reader keyboard shortcuts?

**Comment by sivakumar-kailasam** [comment]
@mansona FYI, I've updated this with master using Github's 'Update branch' button on this PR only to test out the netlify integration.

**Comment by mansona** [comment]
Thanks a million @sivakumar-kailasam 🎉 sorry I've been away for a little while 😖 things have been a bit crazy for me lately.

I'm pretty impressed that this "just works" with the new Ember-cli-docs stuff 🎉 and the navbar is showing up correctly in the demo 😍

**Comment by mansona** [comment]
@MelSumner I'm happy to remove the letter shortcuts 👍 I haven't gotten started with any screen readers for this, maybe we can pair sometime next week and we could walk through it together?

I'm curious why the letter navigation is in the "official" reference implementation if it is going to conflict with existing shortcuts 🤔

**Comment by sivakumar-kailasam** [comment]
>  sorry I've been away for a little while 😖 things have been a bit crazy for me lately.
@mansona Never apologize/feel bad about it. Life happens 🙂I hope things are better now 🙂

**Comment by MelSumner** [comment]
@mansona let's set a time to review, if that works for you? I'd love to help us get this shipped. :+1:

**Comment by MelSumner** [comment]
@mansona would you have time to resolve the conflicts?

**Comment by mansona** [comment]
@MelSumner I've rebased to fix the conflicts 👍 I'm not 100% sure what we need to do to this PR to make it good to go 🤔 I think we still need to solve the mobile view issue

**Comment by MelSumner** [comment]
@mansona ooh look, another chance for you to rebase! :laughing:

**Comment by mansona** [comment]
@MelSumner rebase done 🎉 

Although I think we need to implement the mobile view for the new navbar before we merge this 🤔

File: addon/components/es-navbar/link/component.js [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    'RIGHT': 39,
+    'DOWN': 40
+  }),
+
+  navbar: service(),
+
+  init() {
+    this._super(...arguments);
+
+    this.$.tabIndex = -1;
```
[the remark this answers is not in the document]
Comment (mansona) [reply addon/components/es-navbar/link/component.js]: thanks for the catch @hakilebara 🎉 fixed it

**Comment by mansona** [comment]
Ok so this PR is finally ready for review 🎉 I added a mobile view for the navbar now too. 

Let me know if anyone has any questions 👍

**Pull request merged**